gr-blocks/lib/repeat_impl.h

```diff
@@ -37,6 +39,19 @@ class BLOCKS_API repeat_impl : public repeat
 
     void msg_set_interpolation(const pmt::pmt_t& msg);
     const pmt::pmt_t c_msg_port;
+
+    //! \brief helper function to propagate tags manually
+    void copy_tags(int consumed, int produced, int num_items)
+    {
+        std::vector<tag_t> tags;
+        const auto start_in = nitems_read(0) + static_cast<uint64_t>(consumed);
+        const auto start_out = nitems_written(0) + static_cast<uint64_t>(produced);
+        get_tags_in_range(tags, 0, start_in, start_in + num_items);
+        for (const auto& tag : tags) {
+            const auto offset = d_interp * (tag.offset - start_in) + start_out;
+            add_item_tag(0, offset, tag.key, tag.value);
+        }
+    }
```

Comment:
very nice! I'd put it in the `_impl.cc`, but that's more of a question of preference.

Reply:
My intention in putting it in the `_impl.h` was to have the method inlined in all usages and no "standalone" code for the method. However, this hasn't work as expected. The function doesn't get inlined, and in fact its assembly code is much larger than I expected, so probably it doesn't make much sense inlining it. I have tested what happens if I move the implementation to the `_impl.cc` and declare it `inline`, and the compiler seems to produce the same output. So really no preference from my side here.